tests/queries/0_stateless/01655_plan_optimizations.sh

```diff
@@ -150,3 +150,49 @@ $CLICKHOUSE_CLIENT -q "
     select * from (
         select y, sum(x) from (select number as x, number % 4 as y from numbers(10)) group by y with totals
     ) where y != 2"
+
+echo "> filter is pushed down before CreatingSets"
+$CLICKHOUSE_CLIENT -q "
+    explain select number from (
+        select number from numbers(5) where number in (select 1 + number from numbers(3))
+    ) where number != 2 settings enable_optimize_predicate_expression=0" |
+    grep -o "CreatingSets\|Filter"
```

Comment:
I prefer `grep -e pattern1 -e pattern2` instead of `pattern1|pattern2` for multiple patterns, but it's minor thing.